src/BnbGaugeVotingStateSender.sol

```diff
@@ -0,0 +1,253 @@
+        );
+    }
+
+    /// @notice Sets the recipient for an address on oracle.
+    /// @param _dstChainId Name of the chain.
+    /// @param _recipient Address to set as the recipient.
+    function setRecipient(uint256 _dstChainId, address _recipient) external payable {
+        if (msg.value < setRecipientMinValue) revert InsufficientValue();
+
+        address destinationContract = vms[_dstChainId].claimer;
+
+        if (destinationContract != address(0)) {
+            bytes memory payload = abi.encodeWithSignature("setRecipient(address,address)", msg.sender, _recipient);
+
+            string memory dstChain = vms[_dstChainId].chain;
+            string memory destinationContractHex = destinationContract.toHexStringChecksumed();
+
+            IAxelarGasReceiverProxy(AXELAR_GAS_RECEIVER).payNativeGasForContractCall{value: msg.value}(
+                address(this), dstChain, destinationContractHex, payload, msg.sender
+            );
+
+            IAxelarGateway(AXELAR_GATEWAY).callContract(dstChain, destinationContractHex, payload);
+
+            emit RecipientSet(msg.sender, _recipient, dstChain);
+        }
+    }
+
+    /// @notice Set a xchain vote market info.
+    /// @param _claimer Address of the destination claimer contract.
+    /// @param _dstChain Name of the destination chain.
+    /// @param _dstChainId Destination chain ID.
+    function setVm(address _claimer, string memory _dstChain, uint256 _dstChainId) external onlyGovernance {
+        vms[_dstChainId] = Vm(_claimer, _dstChain);
+
+        emit VmSet(_dstChainId, _claimer, _dstChain);
+    }
+
+    /// @notice Set the min gas value to set the recipient
+    /// @param _newValue min value
+    function setMinValueForSetRecipient(uint256 _newValue) external onlyGovernance {
+        emit RecipientMinValueSet(setRecipientMinValue = _newValue);
+    }
+
+    /// @notice Set the min gas value to claim a bounty
+    /// @param _newValue min value
+    function setClaimMinValue(uint256 _newValue) external onlyGovernance {
+        emit SetClaimerMinValueSet(claimMinValue = _newValue);
+    }
+
+    /// @notice Transfer the governance to a new address.
+    /// @param _governance Address of the new governance.
+    function transferGovernance(address _governance) external onlyGovernance {
+        futureGovernance = _governance;
+    }
+
+    /// @notice Accept the governance transfer.
+    function acceptGovernance() external {
+        if (msg.sender != futureGovernance) revert GovernanceOnly();
+
+        governance = msg.sender;
+
+        /// Reset the future governance.
+        futureGovernance = address(0);
+
+        emit GovernanceChanged(msg.sender);
+    }
+
+    /// @notice Calculates the current period based on weekly intervals
+    /// @return uint256 The start of the current weekly period
+    function getCurrentPeriod() public view returns (uint256) {
```

Comment:
Need to use getCurrentEpoch (2 weeks); 

Since that can cause a disruption if someone claims in the second week of the epoch

Reply:
Done, thanks.